### PR TITLE
Add Exception configuration documentation

### PR DESCRIPTION
**Pull request by dunglas**
Follows #173.

### CODE REVIEW

**Comment by dunglas** [comment]
Ready for review, ping @GregoireHebert

File: core/errors.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+client or validation errors) as well as unexpected errors (PHP exceptions and errors).
+API Platform automatically send the appropriate HTTP status code to the client: `400` for expected errors, `500` for
+unexpected ones. It also provides a description of the respecting [the Hydra specification](http://www.hydra-cg.com/spec/latest/core/#description-of-http-status-codes-and-errors)
+or the [RFC 7807](https://tools.ietf.org/html/rfc7807) depending of the format selected during the [content negotiation](content-negotiation.md).
+
+## Converting PHP Exceptions to HTTP Errors
+
+The framework also allows to configure the HTTP status code sent to the clients when custom exceptions are thrown.
+
+In the following example, we will throw explain to throw a domain exception from the business layer of the application and
```
Comment (GregoireHebert) [inline core/errors.md]: [...] we will throw ~explain to throw~ a domain exception [...]

File: core/errors.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+configure API Platform to convert it to a `404 Not Found` error. Let's create a this domain exception and the service throwing
```
Comment (GregoireHebert) [inline core/errors.md]: [...] Let's create a ~this~ domain exception [...]

File: core/errors.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+    public function checkProductAvailability(GetResponseForControllerResultEvent $event): void
+    {
+        $product = $event->getControllerResult();
+        if (!$product instanceof Product || !$event->getRequest()->isMethodSafe(false)) {
+            return;
+        }
+
+        if (!$product->isPubliclyAvailable()) {
+            // Using internal codes for a better understanding of what's going on
```
Comment (GregoireHebert) [inline core/errors.md]: Finally there isn't internal code used here... Maybe remove the comment ?

File: core/errors.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
+# Error Handling
+
+API Platform comes with a powerful error system. It handles excepted (such as faulty JSON documents sent by the
```
Comment (antograssiot) [inline core/errors.md]: Typo "expected" ?

File: core/errors.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+client or validation errors) as well as unexpected errors (PHP exceptions and errors).
+API Platform automatically send the appropriate HTTP status code to the client: `400` for expected errors, `500` for
+unexpected ones. It also provides a description of the respecting [the Hydra specification](http://www.hydra-cg.com/spec/latest/core/#description-of-http-status-codes-and-errors)
```
Comment (antograssiot) [inline core/errors.md]: "a description respecting" ?

File: core/errors.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+or the [RFC 7807](https://tools.ietf.org/html/rfc7807) depending of the format selected during the [content negotiation](content-negotiation.md).
```
Comment (antograssiot) [inline core/errors.md]: depending on" ?

**Comment by Simperfit** [comment]
This needs a rebase, @GregoireHebert could you please do it ?

**Comment by GregoireHebert** [comment]
Oh sorry no, I can't, this is dunglas branch based on mine... I did it in another PR.
AhAh :sweat_smile: !

**Pull request closed without merging**